framework/SupervisedLearning/KerasRegression.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Created on Dec. 20, 2018
+  @author: wangc
+  base class for tensorflow and keras used for deep neural network
+  i.e. Multi-layer perceptron classifier, CNN, LSTM
+"""
+#for future compatibility with Python 3--------------------------------------------------------------
+from __future__ import division, print_function, unicode_literals, absolute_import
+#End compatibility block for Python 3----------------------------------------------------------------
+
+#External Modules------------------------------------------------------------------------------------
+import copy
+import numpy as np
+import random as rn
+import matplotlib
+import platform
+from scipy import stats
+import os
+import utils.importerUtils
+tf = utils.importerUtils.importModuleLazyRenamed("tf", globals(), "tensorflow")
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from .SupervisedLearning import supervisedLearning
+from .KerasBase import KerasBase
+#Internal Modules End--------------------------------------------------------------------------------
+
+class KerasRegression(KerasBase):
+  """
+    Multi-layer perceptron classifier constructed using Keras API in TensorFlow
+  """
+  ROMType = 'KerasRegression'
+
+  def __init__(self, **kwargs):
+    """
+      A constructor that will appropriately intialize a keras deep neural network object
+      @ In, kwargs, dict, an arbitrary dictionary of keywords and values
+      @ Out, None
+    """
+    super().__init__(**kwargs)
+
+  def readInitDict(self, initDict):
+    """
+      Reads in the initialization dict to initialize this instance
+      @ In, initDict, dict, keywords passed to constructor
+      @ Out, None
+    """
+    super().readInitDict(initDict)
+    self.printTag = 'KerasRegression'
```

Comment:
I suggest to move all self variables to ```__init__``` method

Reply:
Fixed.